core/src/main/java/org/infinispan/container/offheap/BoundedOffHeapDataContainer.java

```diff
@@ -259,6 +259,7 @@ private void ensureSize() {
             }
             try {
                InternalCacheEntry<WrappedBytes, WrappedBytes> ice = offHeapEntryFactory.fromMemory(addressToRemove);
+               passivator.passivate(ice);
```

Comment:
I think if we can move the `passivate` call to after we unlock the `entryWriteLock` would be better. This way we won't block other off heap reads and writes for the same lock while performing the passivation, which can take a while. We will still hold onto the ISPN lock for the given key so it should be fine.

Reply:
We should probably also update https://github.com/vjuranek/infinispan/blob/1398f14e570068daa48c00ef2d4c4b0bcd1c0391/core/src/main/java/org/infinispan/container/offheap/OffHeapDataContainer.java#L499 as well to make that after we unlock the `lock`.

Reply:
@wburns Which ISPN lock? Passivation should not acquire the `LockManager` lock, because regular writes ignore it on the backup owners.

Reply:
Sorry yes there is no lock, in which case we have to use the striped lock from the data container to guarantee consistency.